### ISSUE TITLE
Product Detail Page Improvements for Screen Reader support

### ISSUE THREAD

**Issue by saad-mo** [SAP/spartacus#11194]
- This ticket contains improvements for [Headings and Labels Success Criterion](https://github.com/SAP/spartacus/issues/11192)  ([WCAG 2.1 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html))
- master ticket: #11192

## Labels improvements 

- [x] OUT OF STOCK PRODUCT: Disabled "Notify Me" button does not have keyboard focus, hence it is never announced by Screen Reader. Make the button focusable to fix this issue
- [x] OUT OF STOCK PRODUCT:  Qty "out of stock" note is not keyboard focus. Screen Reader user would not know the product is out of stock  
- [x] Under Reviews, "More/Less" button does not provide context for what the buttons does. Suggestion: aria-label="Load More/Less Reviews" or change button text to be "Load More/Less Reviews" 

## Update / Fix things that break

![Screen Shot 2021-02-16 at 12 47 36 PM](https://user-images.githubusercontent.com/15215400/108421844-5d3de700-71fb-11eb-85b5-db4ced5a5008.png)

![Screen Shot 2021-02-16 at 12 56 24 PM](https://user-images.githubusercontent.com/15215400/108421757-3ed7eb80-71fb-11eb-86f3-026b10575968.png)

**Comment by saad-mo** [comment]
used `aira-describedby` to address issues 1 and 2 from the list.

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

- More Review/Less Review -> OK
- I still don't see keyboard focus for "Notify Me" and "Out of stock"

**Comment by KateChuen** [comment]
but the voiceover reads the text and customer is able to learn it's out of stock if we reloads the page.
 
![Notify1](https://user-images.githubusercontent.com/45409515/145252713-2a2be844-d98b-49fc-bae2-5aece8de7edb.png)
![Notify3](https://user-images.githubusercontent.com/45409515/145252718-88ca8c31-88cb-4742-8a81-848ff7c6b9f3.png)
![Notify2](https://user-images.githubusercontent.com/45409515/145252720-7d22de63-b533-407d-b50c-19f8eb7af446.png)

**Comment by developpeurweb** [comment]
Works with Automatic Read mode, but some things are missing.

@saad-mo it seems the `aria-describedby` changes didn't make it to `develop`.  Also, the "Notify Me" button is missing the blue focus ring in `develop` but works well on `epic/a11y-sr/labels`.

![Screen Shot 2021-12-08 at 12 08 57 PM](https://user-images.githubusercontent.com/15113219/145259035-9d46ada7-0f89-44dc-b1a9-3017bc0aa7b0.png)

![Screen Shot 2021-12-08 at 12 49 35 PM](https://user-images.githubusercontent.com/15113219/145259535-0dd34886-0d6c-4a16-a4f3-27472e800f10.png)

![2021-12-08_13-05-28](https://user-images.githubusercontent.com/15113219/145260549-b9da64ce-bcf7-4d82-95fa-df99cd790e30.jpg)


cc @KateChuen

**Comment by giancorderoortiz** [comment]
Works in mobile as well?

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

OK Now.


As anonymous user:
![Notif2](https://user-images.githubusercontent.com/45409515/157910225-ae973f0d-fff3-4bca-aac2-024ce0dfee82.png)
![Notif3](https://user-images.githubusercontent.com/45409515/157910230-ae08e911-f1f7-4a7d-9125-f3d1d0748768.png)
<img width="1224" alt="Screen Shot 2022-03-11 at 11 53 31 AM" src="https://user-images.githubusercontent.com/45409515/157912022-7d16d62c-5ac9-4a86-890d-45be830fed99.png">

As a registered user:

![Notif4](https://user-images.githubusercontent.com/45409515/157910237-0699b0c4-ccb6-4628-ac33-82272670beb8.png)
![Notif5](https://user-images.githubusercontent.com/45409515/157910242-e45cad02-b21c-42aa-90a6-bd33122272e3.png)
![Notif6](https://user-images.githubusercontent.com/45409515/157910247-363f9f93-d3c7-45f1-aadf-e11ff10b786b.png)

![Notif7](https://user-images.githubusercontent.com/45409515/157910252-38c4f007-39d9-40e3-84cc-c73998baa525.png)

![Notif9](https://user-images.githubusercontent.com/45409515/157910235-75288d10-713e-4084-aa43-872e48b183d3.png)
<img width="1231" alt="Screen Shot 2022-03-11 at 11 52 16 AM" src="https://user-images.githubusercontent.com/45409515/157911991-96abc193-c5f4-4b2e-af47-0fedaf16587a.png">



![Notif8](https://user-images.githubusercontent.com/45409515/157910254-f8b41d6e-aa8c-432c-a1a3-987db4707ecb.png)


To turn notification Off:

![Notif10](https://user-images.githubusercontent.com/45409515/157910223-8a918d20-985d-44c5-b60b-92464d0d3003.png)

![Notif11](https://user-images.githubusercontent.com/45409515/157910218-13c6024b-4249-435f-9fa9-b5f369248787.png)

![Notif13](https://user-images.githubusercontent.com/45409515/157910213-f54c42c7-0218-4a50-9ceb-0a6a3cdd9010.png)
